crates/sui-cluster-test/src/faucet.rs

```diff
@@ -41,7 +43,22 @@ impl FaucetClientFactory {
                 .await
                 .unwrap();
 
-                Arc::new(LocalFaucetClient::new(simple_faucet))
+                let faucet = Arc::new(simple_faucet);
+                let faucet_batch = Arc::clone(&faucet);
+
+                // Spawn async thread here to batch_process_faucet reuuests every 1 second window
+                spawn_monitored_task!(async move {
+                    loop {
+                        // Every 2 seconds we commit faucet requests
+                        tokio::time::sleep(Duration::from_secs(1)).await;
```

Comment:
Related to my comment below about the overall structure of the task that clears the batch queue -- rather than have a sleep here, and a timeout there, you should have a structure as follows, (in pseudo-code):

- Outer loop, where the first step is to wait (without timeout) to `recv` the first request of the batch.
- Once you have that first request, you set a fixed period of time in which to gather the batch (let's say it's 1 second for now), this is handled by calling a function to add to the batch by repeatedly calling `recv` *without a timeout*, but guarding the call to this function with a 1 second timeout.
- At the end of this process, you send off the batch transaction, and repeat (go back to waiting for the first request of the next transaction).

This way, you will not repeatedly poll under low traffic -- you let the tokio runtime tell you when the first request arrives -- and you will also not wait overly long to send out a batch because new requests keep dripping in.  Under high traffic, you will send out batches as fast as they fill the queue.

Reply:
I'm not 100% on this, but I believe a timer with a loop may not work as intended, because the tokio runtime does not know to wake up your task when it times out in that case, whereas with a call to `timeout` it will.  The way I would try to do this with `timeout` is to pass in the vector to accumulate the batch as a `&mut` param to the function.